pkg/cloudprovider/kubevirt/loadbalancer.go

```diff
@@ -223,10 +228,10 @@ func (lb *loadbalancer) createLoadBalancerServicePorts(service *corev1.Service)
 	return ports
 }
 
-func (lb *loadbalancer) applyServiceLabels(lbName, serviceName string, nodes []*corev1.Node) error {
+func (lb *loadbalancer) applyServiceLabels(ctx context.Context, lbName, serviceName string, nodes []*corev1.Node) error {
 	instanceIDs := buildInstanceIDMap(nodes)
-	allVmis, err := lb.kubevirt.VirtualMachineInstance(lb.namespace).List(&metav1.ListOptions{})
-	if err != nil {
+	var allVmis kubevirtv1.VirtualMachineInstanceList
+	if err := lb.client.List(ctx, &allVmis, client.InNamespace(lb.namespace)); err != nil {
 		return fmt.Errorf("Failed to list VMIs: %v", err)
 	}
 	vmiUIDs := make([]string, len(instanceIDs))
```

Comment:
Don't use make, or you will get a slice with a few empty strings inside, this breaks validation later on.
```suggestion
	var vmiUIDs []string
```